### PR TITLE
Remove testing methods from ScheduleAlgorithm

### PR DESCRIPTION
**Pull request by anfernee**
**What this PR does / why we need it**: Simplification. 

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by bsalamat** [comment]
I am not sure about this change. Why is it better than the existing code?

**Comment by anfernee** [comment]
Conceptually, when we think about a scheduler algorithm, for now it's really all about schedule and preempt. Those methods for testing purpose adds assumption like it's `FitPredicate` and `PriorityConfig` related. Those are implementation details. I am not saying you cannot add those methods just for testing purpose, but it's the last resort. When I can avoid doing it, I find a way not to do it.

**Comment by anfernee** [comment]
/sig scheduling
cc @kubernetes/sig-scheduling-pr-reviews

**Comment by bsalamat** [comment]
@anfernee In scheduler code, priorities and predicates are a part of "algorithm". Even their code is located under the "algorithm" directory. That's why I think they should remain in the "algorithm" struct. Otherwise, there is no strong reason for this placement and what you wrote in your previous comment is valid too.

**Comment by anfernee** [comment]
@bsalamat I understand. The thing is the caller of algorithm doesn't really care about priorities and predicates, what it cares is it can do schedule and preempt. The reason those 2 functions in the interface is the testing code needs it. There are other ways to expose it to testing.

File: plugin/pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -144,18 +144,6 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
...
 
-// Prioritizers returns a slice containing all the scheduler's priority
-// functions and their config. It is exposed for testing only.
-func (g *genericScheduler) Prioritizers() []algorithm.PriorityConfig {
-	return g.prioritizers
-}
-
-// Predicates returns a map containing all the scheduler's predicate
-// functions. It is exposed for testing only.
-func (g *genericScheduler) Predicates() map[string]algorithm.FitPredicate {
```
Comment (yastij) [inline plugin/pkg/scheduler/core/generic_scheduler.go]: I'm not sure this change will bring clarity, as we are moving out the problem. To me these are related to scheduler algorithm, we should kept here.

**Review by yastij — CHANGES_REQUESTED** [review]

File: plugin/pkg/scheduler/core/generic_scheduler.go (hunk 1 above)
Comment (resouer) [reply plugin/pkg/scheduler/core/generic_scheduler.go]: Kindly ping @anfernee WDYT? Maybe close this one and focus on the other pending PRs?

Comment (anfernee) [reply plugin/pkg/scheduler/core/generic_scheduler.go]: `// It is exposed for testing only.` this especially worries me. Accumulation of things like these are not going to bring value but harm. but anyway, I am closing this.

**Pull request closed without merging**